src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs

```diff
@@ -257,6 +257,20 @@ private BoundForEachStatement BindForEachPartsWorker(DiagnosticBag diagnostics,
                                     throw ExceptionUtilities.UnexpectedValue(local.RefKind);
                             }
 
+                            if (local.RefKind != RefKind.None)
+                            {
+                                uint collectionRefEscape = GetInvocationEscapeScope(
```

Comment:
I think this will always compute the same value as `collectionValEscape` above that is already computed (since there are no arguments going to either `GetEnumerator`, nor to the `Current`). 

Should just use that as a ref escape for the local.

Reply:
And we probably should just say that in the spec - "ref-escape on a local in a ref-foreach is a val-escape of the collection". 
We are saying something similar in the val case. It is simpler - No need to involve how it is lowered.